connection_configuration_test.go

```diff
@@ -0,0 +1,271 @@
+	testCases := []paramList{
+		{
+			testParams: []string{"user", "password", "host", "account", "warehouse", "database",
+				"schema", "role", "region", "protocol", "passcode", "application", "token",
+				"tracing", "tmpDirPath", "clientConfigFile"},
+			values: []interface{}{"value"},
+		},
+		{
+			testParams: []string{"privatekey"},
+			values:     []interface{}{generatePKCS8StringSupress(testPrivKey)},
+		},
+		{
+			testParams: []string{"port", "maxRetryCount", "clientTimeout", "jwtClientTimeout", "loginTimeout",
+				"requestTimeout", "jwtTimeout", "externalBrowserTimeout"},
+			values: []interface{}{"300", 500},
+		},
+		{
+			testParams: []string{"ocspFailOpen", "insecureMode", "PasscodeInPassword", "validateDEFAULTParameters", "clientRequestMFAtoken",
+				"clientStoreTemporaryCredential", "disableQueryContextCache", "includeRetryReason", "disableConsoleLogin", "disableSamlUrlCheck"},
+			values: []interface{}{true, "true", false, "false"},
+		},
+	}
+
+	for _, testCase := range testCases {
+		for _, param := range testCase.testParams {
+			for _, value := range testCase.values {
+				t.Run(param, func(t *testing.T) {
+					cfg := &Config{}
+					connectionMap := make(map[string]interface{})
+					connectionMap[param] = value
+					err := parseToml(cfg, connectionMap)
+					assertNilF(t, err, "The value should be parsed")
+				})
+			}
+		}
+	}
+}
+
+func TestParseTomlWithWrongValue(t *testing.T) {
+	testCases := []paramList{
+		{
+			testParams: []string{"user", "password", "host", "account", "warehouse", "database",
+				"schema", "role", "region", "protocol", "passcode", "application", "token", "privateKey",
+				"tracing", "tmpDirPath", "clientConfigFile", "wrongParams", "token_file_path"},
+			values: []interface{}{1, false},
+		},
+		{
+			testParams: []string{"port", "maxRetryCount", "clientTimeout", "jwtClientTimeout", "loginTimeout",
+				"requestTimeout", "jwtTimeout", "externalBrowserTimeout", "authenticator"},
+			values: []interface{}{"wrong_value", false},
+		},
+		{
+			testParams: []string{"ocspFailOpen", "insecureMode", "PasscodeInPassword", "validateDEFAULTParameters", "clientRequestMFAtoken",
+				"clientStoreTemporaryCredential", "disableQueryContextCache", "includeRetryReason", "disableConsoleLogin", "disableSamlUrlCheck"},
+			values: []interface{}{"wrong_value", 1},
+		},
+	}
+
+	for _, testCase := range testCases {
+		for _, param := range testCase.testParams {
+			for _, value := range testCase.values {
+				t.Run(param, func(t *testing.T) {
+					cfg := &Config{}
+					connectionMap := make(map[string]interface{})
+					connectionMap[param] = value
+					err := parseToml(cfg, connectionMap)
+					assertNotNilF(t, err, "should have failed")
+					driverErr, ok := err.(*SnowflakeError)
+					assertTrueF(t, ok, "This should be a Snowflake Error")
+					assertEqualF(t, driverErr.Number, ErrCodeTomlFileParsingFailed)
+				})
+
+			}
+		}
+	}
+}
+
+func TestGetTomlFilePath(t *testing.T) {
```

Comment:
It is needed to add test to verify throwing of exception when configuration file (or token file) has permission different than 600.